eclair-core/src/main/scala/fr/acinq/eclair/channel/publish/TxPublisher.scala

```diff
@@ -196,22 +194,21 @@ private class TxPublisher(nodeParams: NodeParams, factory: TxPublisher.ChildFact
         }
 
       case cmd: PublishReplaceableTx =>
-        val targetFeerate = feeEstimator.getFeeratePerKw(feeTargets.commitmentBlockTarget)
         val attempts = pending.getOrElse(cmd.input, Seq.empty)
-        val alreadyPublished = attempts.exists {
-          // If there is already an attempt at spending this outpoint with a higher feerate, there is no point in publishing again.
-          case a: ReplaceableAttempt => targetFeerate <= a.feerate
-          case _ => false
+        val alreadyPublished = attempts.collectFirst {
+          // If there is already an attempt at spending this outpoint with a more aggressive confirmation target, there is no point in publishing again.
```

Comment:
Can this concurrency scenario actually happen?

Reply:
We regularly have commands issued with the same confirmation target as the pending one (whenever the `Channel` calls `doPublish` again because one of the transactions in the `XxxCommitPublished` was confirmed).

But conceptually, the condition is `<=` instead of simply `==`, and it's more future-proof, that's why I used this.

Reply:
There is actually a use-case right now where we may send a command with `target1` and later `target2 > target1`:

- we have a commitment with received htlcs, but don't have preimages for them
- since there's not htlc we can claim, we use a "slow" confirmation target for the commit tx (now + 36 blocks)
- then we receive the preimage for one of the htlcs: the new confirmation target will now be that htlc expiry]
- if that expiry is in more than 36 blocks, the new command we will issue for the commit tx will have a confirmation target bigger than the previous one

Reply:
> we want to keep the previous one which is safer

Just so I'm sure I understand: in this example the previous one isn't actually safer right?

Reply:
It is actually safer, because the commit tx will be confirmed faster, so that you don't need to "rush" the confirmation of the htlc tx (you'll have more blocks between the height where the htlc tx can be broadcast and the height at which it must be confirmed). That extra safety shouldn't be needed though if that's what you mean.